### PR TITLE
Use custom paging for calendar to avoid crashes

### PR DESCRIPTION
**Pull request by vanwagonet**
Also makes plannable date optional to avoid another crash.

[run-nightly]
refs: MBL-14036
affects: Parent
release note: none

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/2x5huq5) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F2x5huq5)
teacher: [Link](https://rebrand.ly/de4ej3l) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fde4ej3l)
parent: [Link](https://rebrand.ly/vxvzvxq) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fvxvzvxq)

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024819&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|13
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-14036](https://instructure.atlassian.net/browse/MBL-14036)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Canvas iOS** | 93.58% | 93.56% | 0.02%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1024840&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|13
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1025004&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|13
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Core/Core/Views/PagesViewController.swift [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+            rightPage = currentPage
+            currentPage = left
+            notifyUpdated()
+        } else if x >= currentPage.view.frame.maxX, let right = rightPage {
+            leftPage?.unembed()
+            rightPage = nil
+            leftPage = currentPage
+            currentPage = right
+            scrollView.contentOffset.x -= scrollView.frame.width
+            targetContentOffset.pointee = CGPoint(x: x - scrollView.frame.width, y: 0)
```
Comment (armstrongnate) [inline Core/Core/Views/PagesViewController.swift]: I think these two calls make sense to me, I'm just curious why you don't have to do them above when swiping right?

**Comment by armstrongnate** [comment]
Code is looking good- easy to follow and reason about. Nice job. I'll install it and take it for a spin.

File: Core/Core/Views/PagesViewController.swift (hunk 1 above)
Comment (vanwagonet) [reply Core/Core/Views/PagesViewController.swift]: You only need to mess with them when you remove the leftmost view (`leftPage?.unembed()`), which only happens in this branch. Dropping off the right view doesn't change where other views are positioned, so the scroll offset doesn't need to change either.

Comment (armstrongnate) [reply Core/Core/Views/PagesViewController.swift]: Ok that makes sense.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1025008&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|13
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Comment by vanwagonet** [comment]
Nightly failure looks unrelated, and I think someone else is addressing this failure caused by renaming a user in iosauto:
> ParentE2ETests/InboxTests/testGetToReplyScreen

**Comment by armstrongnate** [comment]
`testGetToReplyScreen` should have been fixed here https://github.com/instructure/canvas-ios/pull/1135

🤔

**Pull request merged**